### ISSUE TITLE
Models cannot be named `*Data*`

### ISSUE THREAD

**Issue by jneurock** [miragejs/miragejs#782]
Mirage seems to have an issue with model names having `Data` at the end of the name. See [this REPL](https://miragejs.com/repl/v2/47f486405b). Note: It does not seem to matter if the word `Data` appears anywhere else in the model name, e.g., `DataFoo` or `fooDataBar`.

This exception is thrown when trying to create a record of the `*Data` type:
```
Error: Mirage: You called server.create('bar-data') but no model or factory was found.
```

**Comment by jneurock** [comment]
Ok, I figured this out. It has to do with the fact that "data" is plural. See [this REPL](https://miragejs.com/repl/v2/b7bf9076c0). I think this issue could be closed since it does technically work but then 2 gotchas remain:
1. I think, colloquially, people use the word "data" singularly. It was hard for me to figure this out and I'm a native English speaker. I can imagine those with a different native language may struggle with this even more.
2. It seems odd that I can specify a plural model name. A more clear demonstration of that can be seen here in [this REPL](https://miragejs.com/repl/v2/0f71311d8a).

I could see a case for and against addressing this issue, specifically with `*Data` models in the code. It's likely uncommon to have a model name like this and now that the issue has been documented, there's some reference. I could also see a case being made to just call out this case in the docs.

WDYT @samselikoff?

**Comment by samselikoff** [comment]
Yeah woof that's rough. I do hate when libs make landgrabs like this, especially when it's not called out. Extremely frustrating to stumble upon + debug (so thanks for tracking this down!).

Lots of magic in Mirage atm for things like this. I think we could improve upon it in the future but this would likely involve a decent amount of work.

If anyone wants to take it on I'm happy to support in whatever way I can. We can also add a linter/assertion about pluralized model names because I agree that's awkward.

For now, from my side I'm happy to leave it as it is, just because it's not a personal priority. But again I'm happy to support anyone who wants to improve this.
